apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java

```diff
@@ -164,27 +157,35 @@ private void handleEvent(ReportingEvent event, long sequence, boolean endOfBatch
             return;
         }
         processorEventHandler.onEvent(event, sequence, endOfBatch);
-        if (connection == null) {
-            if (gracePeriodEnd > System.currentTimeMillis()) {
-                // back off because there are connection issues with the apm server
-                dropped++;
-                return;
-            }
-            connection = startRequest();
-            payloadSerializer.serializeMetaDataNdJson(metaData);
-        }
         try {
-            writeEvent(event);
-        } catch (Exception e) {
-            // end request on error to start backing off
-            flush();
-            onConnectionError(null, currentlyTransmitting, 0);
+            if (connection == null) {
+                connection = startRequest();
+                payloadSerializer.serializeMetaDataNdJson(metaData);
+            }
+            try {
+                writeEvent(event);
+            } catch (Exception e) {
```

Comment:
I think we can get rid of this try/catch when catching `Exception` instead of `IOException` in the next catch block and by moving `flush()` to that as well. If the connection is not established, `flush()` is a noop anyway.

Reply:
Not the most accurate error handling anyway, but I will improve logging to compensate that